backend/django/core/views/api_annotate.py

```diff
@@ -437,20 +441,50 @@ def data_unlabeled_table(request, project_pk):
     for d in unlabeled_data:
         serialized_data = DataSerializer(d, many=False).data
 
-        sorted_similarity_pairs = []
-        for pair in serialized_data["similarityPair"]:
-            sorted_similarity_pairs.append(pair)
         temp = {
             "Text": serialized_data["text"],
             "metadata": serialized_data["metadata"],
-            "similarityPair": serialized_data["similarityPair"],
             "ID": d.id,
         }
         data.append(temp)
 
     return Response({"data": data})
 
 
+@api_view(["GET"])
+@permission_classes((IsAdminOrCreator,))
+def embeddings_comparison(request, project_pk):
+    """This finds the highest scoring labels when comparing cosine similarity scores of
+    their embeddingsfor a given input string.
+
+    Args:
+        input: The input string
+        request: The POST request
+        project_pk: Primary key of the project
+    Returns:
+        data: a list of data information
+    """
+    project = Project.objects.get(pk=project_pk)
+    project_labels = Label.objects.filter(project=project)
+    project_labels_embeddings = list(
+        LabelEmbeddings.objects.filter(label_id__in=project_labels).values_list(
+            "embedding", flat=True
+        )
+    )
+
+    text = request.GET.get("text")
+    text_embedding = model.encode(text)
+
+    cosine_scores = util.pytorch_cos_sim(text_embedding, project_labels_embeddings)
```

Comment:
So when I start up SMART and look at the annotate page for one of the default projects I get the following in the logs

```
smart_backend     | [21/Dec/2021 16:32:47] "GET /api/embeddings/2?text=when%20the%20hay%20is%20just%20right HTTP/1.1" 301 0
smart_backend     | Internal Server Error: /api/embeddings/2/
smart_backend     | Traceback (most recent call last):
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
smart_backend     |     response = get_response(request)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
smart_backend     |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, smart_backend     | Internal Server Error: /api/embeddings/2/
smart_backend     | Traceback (most recent call last):
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
smart_backend     |     response = get_response(request)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
smart_backend     |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
smart_backend     |     return view_func(*args, **kwargs)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
smart_backend     |     return self.dispatch(request, *args, **kwargs)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 509, in dispatch
smart_backend     |     response = self.handle_exception(exc)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 469, in handle_exception
smart_backend     |     self.raise_uncaught_exception(exc)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
smart_backend     |     raise exc
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 506, in dispatch
smart_backend     |     response = handler(request, *args, **kwargs)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/rest_framework/decorators.py", line 50, in handler
smart_backend     |     return func(*args, **kwargs)
smart_backend     |   File "/code/core/views/api_annotate.py", line 478, in embeddings_comparison
smart_backend     |     cosine_scores = util.pytorch_cos_sim(text_embedding, project_labels_embeddings)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/sentence_transformers/util.py", line 21, in pytorch_cos_sim
smart_backend     |     return cos_sim(a, b)
smart_backend     |   File "/usr/local/lib/python3.8/site-packages/sentence_transformers/util.py", line 42, in cos_sim
smart_backend     |     return torch.mm(a_norm, b_norm.transpose(0, 1))
smart_backend     | RuntimeError: mat1 and mat2 shapes cannot be multiplied (1x384 and 0x1)
```

Reply:
@dsteedRTI still getting this error when looking at default projects

Reply:
I'm still getting this error.

Reply:
Is this a project that was already generated? What do you mean by default projects exactly?

Reply:
I mean those three projects that get created when you rebuild from the call to seed_smart. They are there by default and can be accessed by signing in as root.

Reply:
Was forgetting to ensure the amount of labels > 5... The default projects don't have enough labels so no embeddings were being generated and then the Annotate page was trying to find the embeddings in the database (which did not exist.)

Added a check to make sure there's enough labels for embeddings to be generated and that hopefully should solve this.